packages/astro/src/core/dev/index.ts

```diff
@@ -33,10 +35,20 @@ export default async function dev(config: AstroConfig, options: DevOptions): Pro
 	await options.telemetry.record([]);
 	config = await runHookConfigSetup({ config, command: 'dev' });
 	const { host, port } = config.server;
+
+
+const clientRuntimeScripts = await glob(new URL('../../runtime/client/*.js', import.meta.url).pathname);
+const clientRuntimeFilePaths = clientRuntimeScripts
+.map(script => `astro/client/${path.basename(script)}`)
+// fixes duplicate dependency issue in monorepo when using astro: "workspace:*"
+.filter(filePath => filePath !== 'astro/client/hmr.js');
```

Comment:
Would this be a problem for astro files not in our monorepo, or is `hmr.js` always ready before the client needs it?

Reply:
Yep, it's always ready since we inject `hmr.js` as a hoisted script!